[BUG] @mastra/convex: node:crypto import prevents bundling ConvexStore for Convex's default runtime

## Describe the Bug

`@mastra/convex` cannot be imported from a Convex function running in the default (V8 isolate) runtime — only from `"use node"` actions. Bundling fails on the adapter's `node:crypto` import:

```
✘ [ERROR] Could not resolve "crypto"

    node_modules/@mastra/convex/dist/index.js:5:20:
      5 │ import crypto2 from 'crypto';
```

The only Node API the client adapter uses is `crypto.randomUUID()`, imported as the `node:crypto` module in six files:

- `src/storage/db/index.ts`
- `src/storage/domains/memory/index.ts`
- `src/storage/domains/scores/index.ts`
- `src/storage/domains/schedules/index.ts`
- `src/vector/index.ts`
- `src/vector/native.ts`

Convex's default runtime supports the Web Crypto API globally, including `crypto.randomUUID()` (https://docs.convex.dev/functions/runtimes#supported-apis) — the capability is present, only the module-import form breaks. `node:crypto` is also the sole Node builtin in the adapter's client entrypoint; the `@mastra/convex/server` and `@mastra/convex/schema` entrypoints already bundle cleanly in the default runtime (they are deliberately Node-free so they can be deployed as Convex functions).

## Steps to Reproduce

1. In a Convex project with `@mastra/convex` installed, create a function file **without** `"use node"`:

```ts
// convex/storeProbe.ts
import { actionGeneric } from 'convex/server';
import { ConvexStore } from '@mastra/convex';

export const probe = actionGeneric(async () => typeof ConvexStore);
```

2. Run `npx convex dev` (or `npx convex dev --once --debug-node-apis`).
3. Bundling fails with `Could not resolve "crypto"` and Convex suggests moving the file to the Node.js runtime.

## Expected Behavior

`ConvexStore` (and the vector adapters) are usable from the Convex default runtime. Requiring `"use node"` just to construct the storage client forces Node actions (higher cold starts, no queries/mutations) even though the adapter only speaks HTTP + JSON and every API it needs exists in the isolate.

## Environment

- `@mastra/convex` 1.3.2 (also reproduced on latest main)
- `convex` 1.41.x

--- a/.changeset/lovely-aliens-matter.md
+++ b/.changeset/lovely-aliens-matter.md
@@ -1,0 +1,5 @@
+---
+'@mastra/convex': patch
+---
+
+Removed the Convex adapters' only Node builtin dependency. ConvexStore, the vector adapters, and the storage domains now use the Web Crypto API (crypto.randomUUID) instead of importing node:crypto, so @mastra/convex itself no longer requires the Node.js runtime ("use node") to bundle inside a Convex project.

--- a/stores/convex/src/isolate-safety.test.ts
+++ b/stores/convex/src/isolate-safety.test.ts
@@ -1,0 +1,58 @@
+import { readdirSync, readFileSync } from 'node:fs';
+import { builtinModules } from 'node:module';
+import { join, relative } from 'node:path';
+
+import { describe, expect, it } from 'vitest';
+
+/**
+ * Drift guard: keep the package bundleable for Convex's default (V8 isolate)
+ * runtime, which provides Web APIs but no Node builtin modules.
+ *
+ * The schema and server entrypoints are deployed as Convex functions, and the
+ * client adapters (ConvexStore, vectors, cache) only need HTTP + JSON + Web
+ * Crypto — so no source file may import a Node builtin module. Web-standard
+ * globals (crypto.randomUUID, fetch, ReadableStream) are fine.
+ *
+ * See https://docs.convex.dev/functions/runtimes#supported-apis
+ */
+const SRC_DIR = join(__dirname);
+
+// Builtins Convex's default runtime explicitly provides as importable modules.
+const CONVEX_ALLOWED_BUILTINS = new Set(['async_hooks', 'node:async_hooks']);
+
+function listSourceFiles(dir: string): string[] {
+  return readdirSync(dir, { withFileTypes: true }).flatMap(entry => {
+    const path = join(dir, entry.name);
+    if (entry.isDirectory()) return listSourceFiles(path);
+    if (!entry.name.endsWith('.ts') || entry.name.endsWith('.test.ts')) return [];
+    return [path];
+  });
+}
+
+function findBuiltinImports(source: string): string[] {
+  const specifiers = [...source.matchAll(/(?:from\s+|import\s*\(\s*|require\s*\(\s*)['"]([^'"]+)['"]/g)].map(
+    match => match[1]!,
+  );
+  return specifiers.filter(specifier => {
+    if (CONVEX_ALLOWED_BUILTINS.has(specifier)) return false;
+    const bare = specifier.startsWith('node:') ? specifier.slice(5) : specifier;
+    return specifier.startsWith('node:') || builtinModules.includes(bare);
+  });
+}
+
+describe('Convex isolate-runtime safety', () => {
+  it('no source file imports Node builtin modules', () => {
+    const offenders: string[] = [];
+    for (const file of listSourceFiles(SRC_DIR)) {
+      const builtins = findBuiltinImports(readFileSync(file, 'utf8'));
+      if (builtins.length > 0) {
+        offenders.push(`${relative(SRC_DIR, file)}: ${builtins.join(', ')}`);
+      }
+    }
+    expect(
+      offenders,
+      'Node builtin imports break bundling for the Convex default runtime. ' +
+        'Use Web APIs (e.g. globalThis crypto.randomUUID) or move the code behind an optional Node-only entrypoint.',
+    ).toEqual([]);
+  });
+});

--- a/stores/convex/src/storage/db/index.ts
+++ b/stores/convex/src/storage/db/index.ts
@@ -1,5 +1,3 @@
-import crypto from 'node:crypto';
-
 import { MastraBase } from '@mastra/core/base';
 import type { StorageThreadType } from '@mastra/core/memory';
 import {

--- a/stores/convex/src/storage/domains/memory/index.ts
+++ b/stores/convex/src/storage/domains/memory/index.ts
@@ -1,5 +1,3 @@
-import crypto from 'node:crypto';
-
 import { MessageList } from '@mastra/core/agent';
 import type { MastraMessageContentV2 } from '@mastra/core/agent';
 import { ErrorCategory, ErrorDomain, MastraError } from '@mastra/core/error';

--- a/stores/convex/src/storage/domains/schedules/index.ts
+++ b/stores/convex/src/storage/domains/schedules/index.ts
@@ -1,5 +1,3 @@
-import crypto from 'node:crypto';
-
 import type {
   Schedule,
   ScheduleFilter,

--- a/stores/convex/src/storage/domains/scores/index.ts
+++ b/stores/convex/src/storage/domains/scores/index.ts
@@ -1,5 +1,3 @@
-import crypto from 'node:crypto';
-
 import { ErrorCategory, ErrorDomain, MastraError } from '@mastra/core/error';
 import type {
   ListScoresResponse,

--- a/stores/convex/src/vector/index.ts
+++ b/stores/convex/src/vector/index.ts
@@ -1,5 +1,3 @@
-import crypto from 'node:crypto';
-
 import { MastraError, ErrorDomain, ErrorCategory } from '@mastra/core/error';
 import { createVectorErrorId } from '@mastra/core/storage';
 import { MastraVector } from '@mastra/core/vector';

--- a/stores/convex/src/vector/native.ts
+++ b/stores/convex/src/vector/native.ts
@@ -1,5 +1,3 @@
-import crypto from 'node:crypto';
-
 import { ErrorCategory, ErrorDomain, MastraError } from '@mastra/core/error';
 import { createVectorErrorId } from '@mastra/core/storage';
 import { MastraVector } from '@mastra/core/vector';

